src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
ps I'm ok with the plan of going ahead now with what you have, as long as it doesn't put us too deep into a whole when we need to switch to the more efficient mechanism.  In other words, I don't want us to spend a lot of time writing throw-away code on the current APIs if we're going to switch to something else, but if you believe that there's very little throw-away and most of the code can be saved and repurposed with minimal changes to switch to the revised API set, sounds good.